docs: 1.5 release (#8387)
langflow-ai/langflow: docs/docs/Concepts/concepts-flows.md

slug: /concepts-flows
---

import Icon from "@site/src/components/icon";

Flows in Langflow are fully serializable and can be saved and loaded from the file system. In this guide, we'll explore how to import and export flows.

## Import flow

If you already have a Langflow JSON file on your local machine, from the **Projects** page, click <Icon name="Upload" aria-hidden="true"/> **Upload a flow**.

Once imported, your flow is ready to use.

:::tip
You can drag and drop Langflow JSON files directly from your file system into the Langflow window to import a flow, even into the initial Langflow splash screen.
:::

## Export flow

> 

Reviewer: ```suggestion

You can export flows to transfer flows between Langflow instances or save backups of your flows.

An exported flow is downloaded to your local machine as a JSON file named `FLOW_NAME.json`.

```

To **Export** your flow, in the **Playground**, click **Share**, and then click **Export**.

> Select **Save with my API keys** to save the flow with any **Global variables** included.

Reviewer: ```suggestion
When exporting from the Langflow UI, you can select **Save with my API keys** to export the flow _and_ any defined API key variables.
```

> :::important

Reviewer: ```suggestion
:::warning
```

I would increase this to warning because it potentially exposes the literal key.

If your key is saved as a **Global variable**, only the global variable you created to contain the value is saved.
If your key value is manually entered into a component field, the actual key value is saved in the JSON file.
:::

When you share your flow file with another user who has the same global variables populated, the flow runs without requiring keys to be added again.

> The `FLOW_NAME.json` file is downloaded to your local machine.

Reviewer: ```suggestion
```

See suggestion above to move this to the top of the section.
